osu.Game/Online/Leaderboards/DrawableRank.cs

```diff
@@ -8,67 +8,59 @@
 using osu.Framework.Graphics.Sprites;
 using osu.Framework.Graphics.Textures;
 using osu.Game.Scoring;
+using System;
 
 namespace osu.Game.Online.Leaderboards
 {
-    public class DrawableRank : Container
+    public class DrawableRank : ModelBackedDrawable<ScoreRank>
     {
-        private readonly Sprite rankSprite;
         private TextureStore textures;
 
-        public ScoreRank Rank { get; private set; }
+        public ScoreRank Rank
+        {
+            get => Model;
+            set => Model = value;
+        }
+
+        private ScoreRank rank;
 
         public DrawableRank(ScoreRank rank)
         {
-            Rank = rank;
-
-            Children = new Drawable[]
-            {
-                rankSprite = new Sprite
-                {
-                    RelativeSizeAxes = Axes.Both,
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    FillMode = FillMode.Fit
-                },
-            };
+            this.rank = rank;
         }
 
         [BackgroundDependencyLoader]
-        private void load(TextureStore textures)
+        private void load(TextureStore ts)
         {
-            this.textures = textures;
-            updateTexture();
+            textures = ts ?? throw new ArgumentNullException(nameof(ts));
+            Rank = rank;
         }
 
-        private void updateTexture()
+        protected override Drawable CreateDrawable(ScoreRank rank)
         {
-            string textureName;
+            return new Sprite
+            {
+                RelativeSizeAxes = Axes.Both,
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                FillMode = FillMode.Fit,
+                Texture = textures.Get($"Grades/{getTextureName()}"),
```

Comment:
The `Texture` getter needs to be inside the BDL of a custom sprite, otherwise it's going to run on the update thread. Same with all other implementations here.

Reply:
Done, now `UpdateableRank`, `UpdateableFlag`, `UpdateableAvatar` uses `ModelBackedDrawable`, and each creates its own Drawable, for example: (`UpdateableRank` creates an instance of `DrawableRank`), and each load their texture inside the BDL